app/src/main/java/org/mozilla/fenix/ext/BrowserState.kt

```diff
@@ -9,24 +9,36 @@ import mozilla.components.browser.state.selector.selectedNormalTab
 import mozilla.components.browser.state.state.BrowserState
 import mozilla.components.browser.state.state.TabSessionState
 import mozilla.components.feature.tabs.ext.hasMediaPlayed
+import org.mozilla.fenix.FeatureFlags
+import org.mozilla.fenix.home.recenttabs.RecentTab
+import org.mozilla.fenix.tabstray.browser.TabGroup
+import org.mozilla.fenix.tabstray.browser.maxActiveTime
+import org.mozilla.fenix.tabstray.ext.isNormalTabActiveWithSearchTerm
+import kotlin.math.max
 
 /**
  * Get the last opened normal tab and the last tab with in progress media, if available.
  *
- * @return A list of the last opened tab and the last tab with in progress media
+ * @return A array list of the last opened tab and the last tab with in progress media
  * if distinct and available or an empty list.
  */
-fun BrowserState.asRecentTabs(): List<TabSessionState> {
-    return mutableListOf<TabSessionState>().apply {
+fun BrowserState.asRecentTabs(): ArrayList<RecentTab> {
+    return ArrayList<RecentTab>().apply {
         val lastOpenedNormalTab = lastOpenedNormalTab
         val inProgressMediaTab = inProgressMediaTab
 
-        lastOpenedNormalTab?.let { add(it) }
+        lastOpenedNormalTab?.let { add(RecentTab.Tab(it)) }
 
         if (inProgressMediaTab == lastOpenedNormalTab) {
-            secondToLastOpenedNormalTab?.let { add(it) }
+            secondToLastOpenedNormalTab?.let { add(RecentTab.Tab(it)) }
         } else {
-            inProgressMediaTab?.let { add(it) }
+            inProgressMediaTab?.let { add(RecentTab.Tab(it)) }
+        }
+
+        if (FeatureFlags.tabGroupFeature) {
+            lastSearchGroup()?.let {
+                add(it)
+            }
```

Comment:
```suggestion
            lastSearchGroup()?.let { add(it) }
```
nit: you introduced a different style if you will here compare to add the `add()` right before it

Reply:
Agreed, I'll fix.  Thanks